shell/renderer/api/electron_api_web_frame.cc

```diff
@@ -349,555 +349,640 @@ class SpellCheckerHolder final : public content::RenderFrameObserver {
 
-v8::Local<v8::Value> GetWebPreference(v8::Isolate* isolate,
-                                      v8::Local<v8::Value> window,
-                                      std::string pref_name) {
-  content::RenderFrame* render_frame = GetRenderFrame(window);
-  const auto& prefs = render_frame->GetBlinkPreferences();
-
-  if (pref_name == options::kPreloadScripts) {
-    return gin::ConvertToV8(isolate, prefs.preloads);
-  } else if (pref_name == options::kDisableElectronSiteInstanceOverrides) {
-    return gin::ConvertToV8(isolate,
-                            prefs.disable_electron_site_instance_overrides);
-  } else if (pref_name == options::kBackgroundColor) {
-    return gin::ConvertToV8(isolate, prefs.background_color);
-  } else if (pref_name == options::kOpenerID) {
-    // NOTE: openerId is internal-only.
-    return gin::ConvertToV8(isolate, prefs.opener_id);
-  } else if (pref_name == options::kContextIsolation) {
-    return gin::ConvertToV8(isolate, prefs.context_isolation);
-  } else if (pref_name == options::kWorldSafeExecuteJavaScript) {
-    return gin::ConvertToV8(isolate, prefs.world_safe_execute_javascript);
-  } else if (pref_name == options::kGuestInstanceID) {
-    // NOTE: guestInstanceId is internal-only.
-    return gin::ConvertToV8(isolate, prefs.guest_instance_id);
-  } else if (pref_name == options::kHiddenPage) {
-    // NOTE: hiddenPage is internal-only.
-    return gin::ConvertToV8(isolate, prefs.hidden_page);
-  } else if (pref_name == options::kOffscreen) {
-    return gin::ConvertToV8(isolate, prefs.offscreen);
-  } else if (pref_name == options::kPreloadScript) {
-    return gin::ConvertToV8(isolate, prefs.preload.value());
-  } else if (pref_name == options::kNativeWindowOpen) {
-    return gin::ConvertToV8(isolate, prefs.native_window_open);
-  } else if (pref_name == options::kNodeIntegration) {
-    return gin::ConvertToV8(isolate, prefs.node_integration);
-  } else if (pref_name == options::kNodeIntegrationInWorker) {
-    return gin::ConvertToV8(isolate, prefs.node_integration_in_worker);
-  } else if (pref_name == options::kEnableNodeLeakageInRenderers) {
-    // NOTE: enableNodeLeakageInRenderers is internal-only.
-    return gin::ConvertToV8(isolate, prefs.node_leakage_in_renderers);
-  } else if (pref_name == options::kNodeIntegrationInSubFrames) {
-    return gin::ConvertToV8(isolate, true);
-#if BUILDFLAG(ENABLE_BUILTIN_SPELLCHECKER)
-  } else if (pref_name == options::kSpellcheck) {
-    return gin::ConvertToV8(isolate, prefs.enable_spellcheck);
-#endif
-  } else if (pref_name == options::kPlugins) {
-    return gin::ConvertToV8(isolate, prefs.enable_plugins);
-  } else if (pref_name == options::kEnableWebSQL) {
-    return gin::ConvertToV8(isolate, prefs.enable_websql);
-  } else if (pref_name == options::kWebviewTag) {
-    return gin::ConvertToV8(isolate, prefs.webview_tag);
-  }
-  return v8::Null(isolate);
-}
+ private:
+  bool MaybeGetRenderFrame(const gin_helper::ErrorThrower& thrower,
```

Comment:
slight preference for this to receive `v8::Isolate*` instead of `gin_helper::ErrorThrower`. The latter is a zero-cost wrapper around an isolate pointer and can be "constructed" for free.